enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,885 @@
+---
+title: upgrade-without-registry
+authors:
+- "@jhernand"
+reviewers:
+- "@avishayt"
+- "@danielerez"
+- "@mrunalp"
+- "@nmagnezi"
+- "@oourfali"
+approvers:
+- "@sdodson"
+- "@zaneb"
+- "@LalatenduMohanty"
+api-approvers:
+- "@sdodson"
+- "@zaneb"
+- "@deads2k"
+- "@JoelSpeed"
+creation-date: 2023-06-29
+last-updated: 2023-07-26
+tracking-link:
+- https://issues.redhat.com/browse/RFE-4482
+see-also:
+- https://github.com/openshift/api/pull/1548
+- https://github.com/openshift/machine-config-operator/pull/3839
+- https://issues.redhat.com/browse/OCPBUGS-13219
+- https://github.com/openshift/cluster-network-operator/pull/1803
+replaces: []
+superseded-by: []
+---
+
+# Upgrade without registry
+
+## Summary
+
+Provide an automated mechanism to upgrade a cluster without requiring an image
+registry server, and without requiring OpenShift knowledge for the technicians
+performing the upgrades.
+
+## Motivation
+
+All these stories are in the context of disconnected clusters with limited
+resources, both in the cluster itself and in the surrounding environment:
+
+- The cluster is not connected to the Internet, or the bandwidth is very limited.
+- It isn't possible to bring up additional machines, even temporarily.
+- The resources of the cluster are limited, in particular in terms of CPU and memory.
+- The technicians responsible for performing the upgrade have little or no OpenShift knowledge.
+
+These clusters are usually installed at the customer site by a partner engineer
+collaborating with customer technicians.
+
+Eventually, the cluster will need to be upgraded, and then the technicians will
+need tools that make the process as simple as possible, ideally requiring no
+OpenShift knowledge.
```

Comment:
We are trying to make possible for a technician without any OpenShift knowledge to start the update: she receives a USB stick, plugs it into one of the nodes of the cluster and the rest is automatic. It isn't simple, but we believe it is doable.

I can't justify this in terms of cost, because cost reduction hasn't been our main objective. I acknowledge that at the end of the day everything is a matter of cost, but that was not the main driver here. The main driver was making updates simple where currently they are very inconvenient.

Reply:
It also adds to the point that if update goes wrong then can the technician would be able to collect must-gather or can perform some steps if the workaround for the issue exists.